build.sbt

```diff
@@ -117,12 +117,24 @@ lazy val assemblySettings = List(
   assembly / mainClass := Some("org.broadinstitute.gpp.poolq3.PoolQ")
 )
 
-lazy val publishSettings = List(
-  // Publish to GitHub Packages:
-  githubOwner := "broadinstitute",
-  githubRepository := artifactId,
-  githubTokenSource := TokenSource.Environment("GITHUB_TOKEN") || TokenSource.GitConfig("github.token")
```

Comment:
Could you not just set it to some empty default otherwise? Coming to think of it it's a bit weird that env variable needs to be set in order to build the project.

This might also be an issue within sbt-github-packages. This should be checked lazily when publishing.

Setting "GITHUB_TOKEN" in Scala Steward might be a bit of a hack I think.

Reply:
You're right, it's definitely an issue within `sbt-github-packages`, which absolutely should be loading this value lazily but isn't. It's been noted in issues and even PRs in the repository, but there hasn't been any action there. I assume that `TokenSource` doesn't define a `Literal` instance in the ADT to prevent people from committing sensitive data in their builds.

That's one of the reasons why this PR is appealing to me - GitHub doesn't really support SBT publishing to the GitHub Package Repository anyway, but it does support using Maven, so publishing this way, while less "scala-like", is more in line with GitHub's expectations.

I was working on this a bit during my spare time on vacation, and at the time, I didn't see an easy way to set a default but I might have been looking in the wrong places. I wonder if SBT has a way to do a conditional assignment somehow, will take a quick look.